hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -1474,7 +1512,7 @@ private JobTaskItemByHostMap generateTaskCountByHost(List<HostState> hosts, Iter
         Map<String, Boolean> snapshot = new HashMap<>(recentlyBalancedHosts.asMap());
         for (JobTaskMoveAssignment assignment : candidateAssignments) {
             String newHostID = assignment.getTargetUUID();
-            if (newHostID == null) {
+            if (newHostID == null && !assignment.delete()) {
```

Comment:
Can you explain what this change does?

Reply:
Delete assignments (from purgeMisplacedTasks) have targetUUID = null and assignment.delete() = true. This change prevents pruning delete assignments.
@yuesong reverting this change to an earlier commit because: 
when newHostID == null, assignment.delete() == true. The assignment is pruned by the following check:
`if ((newHost == null) || newHost.hasLive(jobKey) || !canReceiveNewTasks(newHost)) {
                log.warn("[spawn.balancer] decided not to move task from job {} to host {} " +
                         "because it cannot receive the new task", jobID, newHostID);`